commons/ava_networks/high_level_gecko_client.go

```diff
@@ -305,7 +308,7 @@ func (highLevelGeckoClient HighLevelGeckoClient) waitForXchainTransactionAccepta
 	if err != nil {
 		return stacktrace.Propagate(err,"Failed to get status.")
 	}
-	for status != TRANSACTION_ACCEPTED_STATUS {
+	for i := 0; i < highLevelGeckoClient.networkAcceptanceTimeoutInSeconds && status != TRANSACTION_ACCEPTED_STATUS; i++ {
```

Comment:
Separately, did you test that this works? I think there's a bug where, if the timeout is hit, then the `for` loop breaks and it returns a `nil` error which, to the calling function, looks like a success (which is a leftover from this being an infinite loop).

Reply:
good point! implemented different return functions based on the conditions for each waiting function and tested they work appropriately